### ISSUE TITLE
appdatacache.cpp (2078) : Assertion Failed: !bSharedKVSymbols       

### ISSUE THREAD

**Issue by ibrokemypie** [solus-project/linux-steam-integration#8]
On void linux, starting lsi-steam opens the login screen, which after signing in, spews out ``appdatacache.cpp (2078) : Assertion Failed: !bSharedKVSymbols`` and essentially freezes. Have tried running steam --reset before lsi-steam, same thing.

[steam.txt](https://github.com/solus-project/linux-steam-integration/files/1077066/steam.txt)

**Comment by ikeydoherty** [comment]
This is more likely due to the binary compatibility of the host distro, and it's broken somewhere. Just to clarify are you using glibc or musl built packages? Also what drivers, etc.

Lastly, your `os-release` file is apparently missing `VERSION_ID` field.

**Comment by ibrokemypie** [comment]
glibc, just plain intel right now so xf86-video-intel. I think the field is non existent since Void is a rolling release like Arch.

**Comment by ikeydoherty** [comment]
Do you have access to the backtrace? 

```
DEBUGGER=gdb steam
```

**Comment by ibrokemypie** [comment]
is this what you meant?
```
$ DEBUGGER=gdb lsi-steam
/extra/home/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
/extra/home/.local/share/Steam/steam.sh: line 154: VERSION_ID: unbound variable
Running Steam on void  64-bit
STEAM_RUNTIME is disabled by the user
GNU gdb (GDB) 8.0
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-unknown-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /extra/home/.local/share/Steam/ubuntu12_32/steam...(no debugging symbolsfound)...done.
LD_PRELOAD = /usr/$LIB/libstdc++.so.6 /usr/$LIB/libgcc_s.so.1 /usr/$LIB/libxcb.so.1
(gdb)                     
```

**Comment by ikeydoherty** [comment]
Alright so you need to tell it to `run`, and when it crashes, use `bt` to get the backtrace

**Comment by ibrokemypie** [comment]
Did do, updated the comment as such

**Comment by ikeydoherty** [comment]
Looking at https://github.com/ValveSoftware/steam-for-linux/issues/5014 this very much looks like an issue with your host libraries

**Comment by ikeydoherty** [comment]
Sidenote also make sure you're using the latest Steam package too so that Steam itself isn't using problematic stuff

**Comment by ibrokemypie** [comment]
I have completely removed steam, updated the package cache, deleted ~/.steam and then reinstalled steam, error persists. How would I diagnose which library is to blame?

**Comment by Vaelatern** [comment]
Would `libappindicator` being built against gtk+3 be a possible problem?

**Comment by ikeydoherty** [comment]
Ideally you want both gtk2 and gtk2 builds of that available. See how we tricksed it in Solus for an example: https://dev.solus-project.com/source/libappindicator/browse/master/package.yml

Plus the emul32 patch to save needing emul32 python: https://dev.solus-project.com/source/libappindicator/browse/master/files/0001-Disable-Python-build-required-for-multilib-build.patch

Same thing for libdbusmenu https://dev.solus-project.com/source/libdbusmenu/browse/master/package.yml

**Comment by ikeydoherty** [comment]
Also given you're using the open source drivers and this looks suspiciously like some kind of race condition, it might be worth trying to disable DRI3 in mesa too to see if it helps:

`export LIBGL_DRI3_DISABLE=1`

**Comment by ibrokemypie** [comment]
export LIBGL_DRI3_DISABLE=1 has made no difference from what I can tell, exactly the same.
something that is possibly of note, running ``STEAM_RUNTIME_PREFER_HOST_LIBRARIES=0 STEAM_RUNTIME=1 steam`` results in ``Failed to load steamui.so - dlerror(): /usr/lib32/libXxf86vm.so.1: undefined symbol: _XEatDataWords                           ``

**Comment by ikeydoherty** [comment]
Please re-try with 0.4

**Comment by QkiZMR** [comment]
```
Running Steam on ubuntu 17.04 64-bit
STEAM_RUNTIME is disabled by the user
[2017-10-11 23:22:20] Startup - updater built Oct  4 2017 22:37:14
[2017-10-11 23:22:21] Weryfikowanie instalacji...
[2017-10-11 23:22:21] Verification complete
appdatacache.cpp (2095) : Assertion Failed: !bSharedKVSymbols
roaming config store loaded successfully - 4802 bytes.
migrating temporary roaming config store
appdatacache.cpp (2095) : Assertion Failed: !bSharedKVSymbols
appdatacache.cpp (2095) : Assertion Failed: !bSharedKVSymbols
appdatacache.cpp (2095) : Assertion Failed: !bSharedKVSymbols
```
Same problem on Ubuntu 17.04 after enabling native system libs in lsi-settings tool.

**Comment by ikeydoherty** [comment]
Truncating it won't help me :) I need the full log data.

This will result in a very large log, but you could also do `LD_DEBUG=all lsi-steam` and upload the (very large) log somewhere so I can run through it.

**Comment by QkiZMR** [comment]
I truncated repeating error appdatacache.cpp. Today I will run with ld_debug var and will send results.

**Comment by QkiZMR** [comment]
liblsi-intercept enabled, rest is disabled.

**Comment by QkiZMR** [comment]
native runtime provided by OS enabled, rest is disabled. Steam doesn't start. Freeze on connecting window.
[native runtime.log](https://github.com/solus-project/linux-steam-integration/files/1384668/native.runtime.log)

**Comment by norg** [comment]
I have the same issue with 0.6 on 64 Bit Archlinux. The logfile would be 287MB are you sure you want the whole debug log or should we focus on specific parts? :)

**Comment by ikeydoherty** [comment]
This is very likely an issue with trying to load `steamui.so`, try:

```
LSI_DEBUG=1 LD_AUDIT="/usr/\$LIB/liblsi-intercept.so" ldd steamui.so
```

You'll need to do this in your Steam ubuntu12_32 directory, i.e. `~/.local/share/Steam/ubuntu12_32`

**Comment by norg** [comment]
Looks like I need to find out which libs are in which packages :)
```
[andi@x250 ubuntu12_32]% LSI_DEBUG=1 LD_AUDIT="/usr/\$LIB/liblsi-intercept.so" ldd steamui.so
	linux-gate.so.1 (0xf7f44000)
	libtier0_s.so => not found
	libvstdlib_s.so => not found
	libv8.so => not found
	libvideo.so => not found
```

**Comment by ikeydoherty** [comment]
Those are all within same directory

**Comment by norg** [comment]
Even when I add those to the `$LD_LIBRARY_PATH` and the `ldd` output looks fine then, I still run into the described issue above :/

**Comment by norg** [comment]
Do you think the pkbuild for arch needs to handle those libs? https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=linux-steam-integration

**Comment by lee8oi** [comment]
Whatever this issue is, still isn't resolved in the Arch AUR linux-steam-integration package yet. Sad panda. But I'm cool with being patient.

**Comment by ikeydoherty** [comment]
At this point I'm gonna start focusing on moving us to the snap instead of trying to package LSI up - it's incredibly hard for the distros (as you're seeing!)

@lee8oi i reckon adding those missing deps would be a very good move :)

**Comment by norg** [comment]
Ah thanks @lee8oi that helped :) I installed the native runtime as well and now steam is starting with LSI. Although Project Highrise still doesn't work. Will open another issue for that.

**Comment by ikeydoherty** [comment]
Yeah its working here so I'm still not convinced you have it installed/activated right.

**Comment by corusm** [comment]
Same issue in my case. Is there any fix yet?
`/data/src/clientdll/appdatacache.cpp (2538) : Assertion Failed: !bSharedKVSymbols`

**Comment by deathxxx123** [comment]
> Same issue in my case. Is there any fix yet?
> `/data/src/clientdll/appdatacache.cpp (2538) : Assertion Failed: !bSharedKVSymbols`

Same issue(maybe):
src/clientdll/appdatacache.cpp (2679) : Assertion Failed: !bSharedKVSymbols
Manjaro Linux Gnome + Nvidia 1060 6GB

I found solution here:
https://bugs.archlinux.org/task/71092#comment200392

**Comment by OnlyCS** [comment]
Install `lib32-libnm` for Arch Linux, would recommend adding it as an AUR dep.

**Comment by onlycs** [comment]
`cd` into `~/.steam/root/ubuntu12_32`, then run:
```
pushd ~/.steam/root/ubuntu12_32; file * | grep ELF | cut -d: -f1 | LD_LIBRARY_PATH=. xargs ldd | grep 'not found' | sort | uniq; popd
```

this will list all missing libraries. **install them!**
I had to install `lib32-libudev0-shim` (for libudev.so.0) and `lib32-libnm` (for libnm.so.0)
`sudo pacman -Fy somelib.so.0` for finding packages
